src/main/java/edu/iu/terracotta/service/app/impl/MediaServiceImpl.java

```diff
@@ -0,0 +1,186 @@
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MediaServiceImpl.class);
+
+    public static final String DATA_VERSION = "http://purl.imsglobal.org/ctx/caliper/v1p2";
+
+    @Autowired
+    private AllRepositories allRepositories;
+
+    @Autowired
+    private SubmissionService submissionService;
+
+    @Autowired
+    private APIJWTService apijwtService;
+
+
+    private String applicationName;
+    private String applicationUrl;
+    private final SoftwareApplication softwareApplication;
+    private final JsonldContext context;
+
+    @Autowired
+    public MediaServiceImpl(@Value("${application.name}") final String applicationNameAttribute,
+                            @Value("${application.url}") final String applicationUrlAttribute) {
+        applicationName = applicationNameAttribute;
+        applicationUrl = applicationUrlAttribute;
+        context = JsonldStringContext.create(DATA_VERSION);
+        softwareApplication = Utils.prepareSoftwareApplication(applicationName, applicationUrl);
+    }
+
+    @Override
+    public MediaEventDto toDto(Event mediaEvent) {
+        MediaEventDto mediaEventDto = new MediaEventDto();
+        PersonDto personDto = new PersonDto();
+        GroupDto groupDto = new GroupDto();
+        MediaObjectDto mediaObjectDto = new MediaObjectDto();
+        MediaLocationDto mediaLocationDto = new MediaLocationDto();
+        MembershipDto membershipDto = new MembershipDto();
+        SessionDto sessionDto = new SessionDto();
+
+        mediaEventDto.setContext(mediaEvent.getLtiContextId());
+        mediaEventDto.setType(EventType.valueOf(mediaEvent.getType()));
+        mediaEventDto.setAction(Action.valueOf(mediaEvent.getAction()));
+        mediaEventDto.setProfile(mediaEvent.getProfile());
+        mediaEventDto.setId(mediaEvent.getCaliperId());
+
+        personDto.setType(EntityType.valueOf(mediaEvent.getActorType()));
+        personDto.setId(mediaEvent.getActorId());
+        mediaEventDto.setActor(personDto);
+
+        groupDto.setId(mediaEvent.getGroup());
+        mediaEventDto.setGroup(groupDto);
+
+        mediaObjectDto.setId(mediaEvent.getObjectId());
+        mediaObjectDto.setMediaType(mediaEvent.getObjectType());
+        mediaEventDto.setObject(mediaObjectDto);
+
+        mediaLocationDto.setId(mediaEvent.getTargetId());
+        mediaLocationDto.setType(EntityType.valueOf(mediaEvent.getTargetType()));
+        mediaEventDto.setTarget(mediaLocationDto);
+
+        membershipDto.setId(mediaEvent.getMembershipId());
+        membershipDto.setRoles(new String[]{mediaEvent.getMembershipRoles()});
+        mediaEventDto.setMembership(membershipDto);
+
+        sessionDto.setId(mediaEvent.getFederatedSession());
+        mediaEventDto.setSession(sessionDto);
+
+        return mediaEventDto;
+    }
+
+    @Override
+    public edu.iu.terracotta.model.events.Event fromDto(MediaEventDto mediaEventDto, SecuredInfo securedInfo,
```

Comment:
The methods in this class are probably better off in CaliperServiceImpl where a lot of these are already implemented (like creating an LTISession object), and we want to use the same methods so that these common fields in all events are formatted the same.

Also, there are two Caliper settings that govern whether to send events to a Caliper endpoint and whether to store them in a database. See for example CaliperServiceImpl.sendAssignmentSubmitted().

Reply:
Thanks for pointing out that @machristie . I have updated the PR